### PR TITLE
[efr32] add BRD4304A (MGM12P) support and board-specific hal-config.h

### PR DESCRIPTION
**Pull request by tomas-c**
Copied existing hal-config.h for BRD4161A and BRD4166A. For BRD4304A
used hal-config from Silabs Thread 2.6.2.0 ncp-uart-hw project without
WDOG, VUART, VCOM, UARTNCP and BSP_SERIAL_APP_PORT settings.

### CODE REVIEW

**Comment by tomas-c** [comment]
Haven't tested examples/platforms/efr32/brd4304a/hal-config.h apart from the fact that it compiles. Will do so soon. Just wanted to see what you all think about having board-specific hal-config.h?

**Comment by jwhui** [comment]
@tomas-c, I think it makes sense to have board-specific `hal-config.h` files - they are board-specific by definition after all.

**Comment by tomas-c** [comment]
Looks like MGM12P and possibly some other boards require FEM to be enabled. I have pushed changed that do that but we are still testing if anything else needs to change to make it work with MGM12P.

**Comment by tomas-c** [comment]
@jwhui This has been tested and works with MGM12P - can create a network and ping each other. I'm happy for this to be merged if no one objects.

File: examples/platforms/efr32/Makefile.am [hunk 1]
```diff
@@ -67,6 +67,7 @@ libopenthread_efr32_a_CPPFLAGS                                                =
     -I$(EFR32MG_SDK_SRCDIR)/platform/emdrv/rtcdrv/inc                           \
     -I$(EFR32MG_SDK_SRCDIR)/platform/emlib/inc                                  \
     -I$(EFR32MG_SDK_SRCDIR)/platform/halconfig/inc/hal-config                   \
+    -I$(EFR32MG_SDK_SRCDIR)/util/plugin/plugin-common/fem-control/              \
```
Comment (jwhui) [inline examples/platforms/efr32/Makefile.am]: nit: remove trailing `/`?

**Comment by jwhui** [comment]
@tomas-c, thanks for testing and reporting back!

**Comment by tomas-c** [comment]
@jwhui pushed a fix following your review, tell me if anything else needs changing

**Pull request merged**